src/components/inputs/SupplierTable.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Supplier } from '../data/data';
+
+interface SupplierTableProps {
+  suppliers: Supplier[];
+  selectSupplierName: string | null;
+  onSelectSupplier: (name: string) => void;
+}
+
+const SupplierTable: React.FC<SupplierTableProps> = ({
+  suppliers,
+  selectSupplierName,
+  onSelectSupplier,
+}) => {
+  return (
+    <div className="suppliers-results-container">
+      <div className="top-bar-title-container">
+        <div className="top-bar-title">▼ Supplier list</div>
+      </div>
+      <table>
+        <thead>
+          <tr>
+            <th></th>
+            <th>Supplier name</th>
+            <th>Supplier index</th>
+            <th>City</th>
+          </tr>
+        </thead>
+        <tbody>
+          {suppliers.map((supplier) => (
+            <tr key={supplier.supplierIndex}>
+              <td>
+                <input
+                  type="radio"
+                  name="supplier"
+                  checked={supplier.name === selectSupplierName}
+                  onChange={() => onSelectSupplier(supplier.name)}
```

Comment:
Here also, do not use name as identifier as it is not unique. You allways need to find unique property. In the moest cases it will be ID but for supplier INDEX is actually unique and can be used as ID